lib/screens/surveydetails/widget/question_paging_widget.dart

```diff
@@ -28,80 +30,96 @@ class _QuestionPagingWidgetState extends State<QuestionPagingWidget> {
+              .image,
+        ),
         SafeArea(
           child: Padding(
             padding: const EdgeInsets.all(20),
             child: Column(
               crossAxisAlignment: CrossAxisAlignment.start,
               children: [
-                _buildQuestionToolbar(context, question),
-                const SizedBox(height: Dimensions.paddingLarge),
-                _buildQuestionHeader(context, question),
+                _buildQuestionToolbar(
+                  context,
+                  widget.surveyDetailUiModel.questions[_currentIndex],
+                ),
+                _pagedQuestions(context, pageController),
                 const Spacer(),
-                _buildQuestionFooter(context, question, controller),
+                _buildQuestionFooter(
+                  context,
+                  widget.surveyDetailUiModel.questions[_currentIndex],
+                  pageController,
+                ),
               ],
             ),
           ),
-        )
+        ),
       ],
     );
   }
 
+  Widget _pagedQuestions(
+    BuildContext context,
+    PageController pageController,
+  ) {
+    return Expanded(
+      child: PageView(
+        controller: pageController,
+        onPageChanged: _onPageChanged,
+        physics: const NeverScrollableScrollPhysics(),
+        children: widget.surveyDetailUiModel.questions
+            .map((QuestionUiModel question) =>
+                _buildQuestionHeader(context, question))
+            .toList(),
+      ),
+    );
+  }
+
   Widget _buildQuestionToolbar(BuildContext context, QuestionUiModel question) {
-    return Row(
-      children: [
-        _isIntroQuestion(question)
-            ? MaterialButton(
-                onPressed: () => {context.pop()},
-                minWidth: 30,
-                height: 30,
-                materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
-                padding: EdgeInsets.zero,
-                child: SvgPicture.asset(Assets.images.icBack),
-              )
-            : const SizedBox(),
-        const Spacer(),
-        !_isIntroQuestion(question)
-            ? ElevatedButton(
-                onPressed: () => {context.pop()},
-                style: ElevatedButton.styleFrom(
-                  shape: const CircleBorder(),
-                  padding: const EdgeInsets.all(8.5),
-                  backgroundColor: Colors.white.withOpacity(0.2),
-                ),
-                child: Icon(
-                  Icons.close,
-                  color: Theme.of(context).primaryColor,
-                ),
-              )
-            : const SizedBox(),
-      ],
+    return Padding(
+      padding: const EdgeInsets.only(bottom: Dimensions.paddingMedium),
+      child: SizedBox(
+        height: 44.0,
+        child: Row(
+          children: [
+            _isIntroQuestion(question)
+                ? MaterialButton(
+                    onPressed: () => {context.pop()},
```

Comment:
when using `() => {}`, it means we returns a map object `{}` instead of calling a function 😄 
Only these two approaches are acceptable:
- `() => abc()`
- `() { abc(); }`
```suggestion
                    onPressed: () => context.pop(),
```